infrastructure/cli/commands/redis/index.ts

```diff
@@ -46,32 +62,18 @@ export class Client {
 
   async connectToRedisLocally(
     port: Port,
-    callback: (redisClient: redis.RedisClient) => void | Promise<void>,
+    callback: (redisClient: RedisClientType) => void | Promise<void>,
   ): Promise<void> {
-    const portNum = typeof port === 'string' ? parseInt(port, 10) : port
-    process.stderr.write(`Connecting to redis on local port ${portNum}…\n`)
-    const redisClient = redis.createClient({
-      password: this.authToken,
-      host: 'localhost',
-      port: portNum,
-      tls: {
-        requestCert: false,
-        rejectUnauthorized: false,
-      },
-    })
+    const redisClient = createLocalRedisClient(port, this.authToken)
+
+    process.stderr.write(`Connecting to redis on local port ${port}…\n`)
+    await redisClient.connect()
     const value = callback(redisClient)
     if (value && 'then' in value) {
       await value
     }
-    return new Promise((resolve, reject) => {
-      process.stderr.write('Redis quitting…\n')
-      redisClient.quit((err) => {
-        if (err) {
-          reject(err)
-        } else {
-          resolve()
-        }
-      })
```

Comment:
Just double checking, the `redisClient.on('error', err => throw err)` redundant in this case right?

```javascript
  redisClient.on('error', (err) => {
    throw err
  })
```

because if the `quit()` promise is rejected it would throw an error and bubble up to this function caller right?

Reply:
don't really know. the client still has the error event listener hook in v4. but if all methods are async and therefore throw exceptions… in what situation would you want to add an event listener? i figured since the docs mention it, i'll just leave it in like in the v3 form.

NB: it's not only quit that can error (and therefore throw here), stuff inside the callback can too

Reply:
I _think_ that `redisClient.on('error', err => throw err)` may be redundant because we're using `await` and the code would behave like synchronous code, e.g. it would throw an error if the promise is rejected which would then reject this promise and bubble up. 

So I _think_ in this case (cause we're using `await`) it shouldn't be necessary, but maybe I'm wrong!

Reply:
true, this `redisClient` is never passed off to third-parties so we can rely on errors thrown at each await point (which we control)